app/main.py

```diff
@@ -1,3 +1,74 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: float) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(
+                km=self.km + other
+            )
+
+        return Distance(
+            km=self.km + other.km
+        )
+
+    def __iadd__(self, other: float) -> Distance:
```

Comment:
The `__iadd__` method should handle in-place addition with both `Distance` instances and numeric types. Similar to `__add__`, it should ensure `other` is a numeric type when it's not a `Distance`.